tests/numeric/test_BaseNumericTransformer.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+import tests.test_data as d
+from tests.base_tests import GenericFitTests, GenericInitTests, GenericTransformTests
+
+
+class BaseNumericTransformerInitTests(GenericInitTests):
+    """
+    Tests for BaseNumericTransformer.init().
+    Note this deliberately avoids starting with "Tests" so that the tests are not run on import.
+    """
+
+
+class BaseNumericTransformerFitTests(GenericFitTests):
+    """
+    Tests for BaseNumericTransformer.fit().
+    Note this deliberately avoids starting with "Tests" so that the tests are not run on import.
+    """
+
+    def test_non_numeric_exception_raised(self, initialized_transformers):
+        """Test an exception is raised if self.columns are non-numeric in X."""
+        df = d.create_df_2()
```

Comment:
Q: Why are we not using the minimal dataframe logic in these tests?

Reply:
Is it because you need specific column types in these, e.g. a string column to raise exception here?

Reply:
yes that's right! could use min dataframe and then overwrite a column to be str, but this seemed easier

Reply:
Cool, agreed :)